src/bitcoin.cpp

```diff
@@ -0,0 +1,198 @@
+
+Commands:
+  gui [ARGS]     Start GUI, equivalent to running 'bitcoin-qt [ARGS]' or 'bitcoin-gui [ARGS]'.
+  node [ARGS]    Start node, equivalent to running 'bitcoind [ARGS]' or 'bitcoin-node [ARGS]'.
+  rpc [ARGS]     Call RPC method, equivalent to running 'bitcoin-cli -named [ARGS]'.
+  wallet [ARGS]  Call wallet command, equivalent to running 'bitcoin-wallet [ARGS]'.
+  tx [ARGS]      Manipulate hex-encoded transactions, equivalent to running 'bitcoin-tx [ARGS]'.
+  help [-a]      Show this help message. Include -a or --all to show additional commands.
+)";
+
+static constexpr auto HELP_EXTRA = R"(
+Additional less commonly used commands:
+  bench [ARGS]      Run bench command, equivalent to running 'bench_bitcoin [ARGS]'.
+  chainstate [ARGS] Run bitcoin kernel chainstate util, equivalent to running 'bitcoin-chainstate [ARGS]'.
+  test [ARGS]       Run unit tests, equivalent to running 'test_bitcoin [ARGS]'.
+  test-gui [ARGS]   Run GUI unit tests, equivalent to running 'test_bitcoin-qt [ARGS]'.
+)";
+
+struct CommandLine {
+    bool use_multiprocess{false};
+    bool show_version{false};
+    bool show_help{false};
+    bool show_help_all{false};
+    std::string_view command;
+    std::vector<const char*> args;
+};
+
+CommandLine ParseCommandLine(int argc, char* argv[]);
+static void ExecCommand(const std::vector<const char*>& args, std::string_view argv0);
+
+int main(int argc, char* argv[])
+{
+    try {
+        CommandLine cmd{ParseCommandLine(argc, argv)};
+        if (cmd.show_version) {
+            tfm::format(std::cout, "%s version %s\n%s", CLIENT_NAME, FormatFullVersion(), FormatParagraph(LicenseInfo()));
+            return EXIT_SUCCESS;
+        }
+
+        std::vector<const char*> args;
+        if (cmd.show_help || cmd.command.empty()) {
+            tfm::format(std::cout, HELP_USAGE, argv[0]);
+            if (cmd.show_help_all) tfm::format(std::cout, HELP_EXTRA);
+            return cmd.show_help ? EXIT_SUCCESS : EXIT_FAILURE;
+        } else if (cmd.command == "gui") {
+            args.emplace_back(cmd.use_multiprocess ? "qt/bitcoin-gui" : "qt/bitcoin-qt");
+        } else if (cmd.command == "node") {
+            args.emplace_back(cmd.use_multiprocess ? "bitcoin-node" : "bitcoind");
+        } else if (cmd.command == "rpc") {
+            args.emplace_back("bitcoin-cli");
+            args.emplace_back("-named");
+        } else if (cmd.command == "wallet") {
+            args.emplace_back("bitcoin-wallet");
+        } else if (cmd.command == "tx") {
+            args.emplace_back("bitcoin-tx");
+        } else if (cmd.command == "bench") {
+            args.emplace_back("bench/bench_bitcoin");
+        } else if (cmd.command == "chainstate") {
+            args.emplace_back("bitcoin-chainstate");
+        } else if (cmd.command == "test") {
+            args.emplace_back("test/test_bitcoin");
+        } else if (cmd.command == "test-gui") {
+            args.emplace_back("qt/test/test_bitcoin-qt");
```

Comment:
are these prepended paths (here and also for the `gui`, `bench` and `test` commands above) still relevant? They seem to reflect a folder structure that doesn't exist anymore since https://github.com/bitcoin/bitcoin/pull/31161